lib/storage/file_storage.ex

```diff
@@ -55,19 +55,23 @@ defmodule Uploadex.FileStorage do
   @impl true
   def get_temporary_file(%{filename: filename}, path, opts), do: get_temporary_file(filename, path, opts)
   def get_temporary_file(filename, path, opts) when is_binary(filename) do
-    delay = Keyword.get(opts, :delete_after, 30_000)
+    delay = Keyword.get(opts, :delete_after, :timer.seconds(30))
     full_path = get_file_full_path(opts, filename)
 
     destination_file = Ecto.UUID.generate() <> Path.extname(filename)
     destination_path = Path.join(path, destination_file)
 
     File.cp!(full_path, destination_path)
     delete_file_after_delay(delay, destination_path)
+
+    destination_path
   end
 
-  def delete_file_after_delay(delay, path) do
-    TaskAfter.task_after(delay, fn -> File.rm(path) end)
-    path
+  defp delete_file_after_delay(delay, path) do
+    Task.start(fn ->
+      Process.sleep(delay)
+      File.rm(path)
+    end)
```

Comment:
I'm used to call `:timer.sleep/1` to do delays, and `:timer` module has other useful functions like [`:timer.apply_after/4`](https://www.erlang.org/doc/man/timer#apply_after-4) that wait how long you ask it for and then spaws a process to execute the function.

```elixir
:timer.apply_after(delay, File, :rm, path)
```